pkg/apis/navigator/v1alpha1/types.go

```diff
@@ -32,38 +32,58 @@ type CassandraCluster struct {
 type CassandraClusterSpec struct {
 	NavigatorClusterConfig `json:",inline"`
 
-	NodePools []CassandraClusterNodePool `json:"nodePools"`
+	// List of node pools belonging to this CassandraCluster.
+	// NodePools cannot currently be removed.
+	// A cluster with 0 node pools will not function correctly.
+	NodePools []CassandraClusterNodePool `json:"nodePools,omitempty"`
```

Comment:
Not sure about this. Would an empty list be so bad? Clearer I'd say, than the attribute just not being there.
What if I need to write a jsonpath query to pick an item from that list? And the list just isn't there in the json representation.

Reply:
Lists and maps render to `null` if not set. This can break some clients, and looks ugly in `kubectl `output (and also is weird from a JSON point of view).

We have to include `omitempty` unless we have some form of validation that ensures that there's always at least one thing in the array basically. We don't currently have that check as part of our validation. Perhaps we should, but I didn't want to make that decision as part of this PR.